internal/bminventory/inventory_test.go

```diff
@@ -5717,6 +5719,145 @@ var _ = Describe("KubeConfig download", func() {
 	})
 })
 
+var _ = Describe("GetMinimalInitrd", func() {
+	var (
+		bm        *bareMetalInventory
+		cfg       Config
+		db        *gorm.DB
+		ctx       = context.Background()
+		clusterID strfmt.UUID
+		infraEnv  common.InfraEnv
+		dbName    string
+	)
+
+	BeforeEach(func() {
+		db, dbName = common.PrepareTestDB()
+		clusterID = strfmt.UUID(uuid.New().String())
+		bm = createInventory(db, cfg)
+		infraEnv = common.InfraEnv{
+			InfraEnv: models.InfraEnv{
+				ID:            clusterID,
+				PullSecretSet: true,
+				Type:          models.ImageTypeMinimalIso,
+			},
+			PullSecret: "{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}",
+		}
+		err := db.Create(&infraEnv).Error
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, dbName)
+		ctrl.Finish()
+	})
+
+	It("returns not found with a non-existant cluster", func() {
+		params := installer.GetMinimalInitrdParams{InfraEnvID: strfmt.UUID(uuid.New().String())}
+		response := bm.GetMinimalInitrd(ctx, params)
+		verifyApiError(response, http.StatusNotFound)
+	})
+
+	It("returns conflict when not minimal-iso", func() {
+		clusterID = strfmt.UUID(uuid.New().String())
+		infraEnv = common.InfraEnv{
+			InfraEnv: models.InfraEnv{
+				ID:            clusterID,
+				PullSecretSet: true,
+				Type:          models.ImageTypeFullIso,
+			},
+			PullSecret: "{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}",
+		}
+		err := db.Create(&infraEnv).Error
+		Expect(err).ShouldNot(HaveOccurred())
+
+		params := installer.GetMinimalInitrdParams{InfraEnvID: clusterID}
+		response := bm.GetMinimalInitrd(ctx, params)
+		verifyApiError(response, http.StatusConflict)
+	})
+
+	It("returns no content without network customizations", func() {
+		params := installer.GetMinimalInitrdParams{InfraEnvID: clusterID}
+		response := bm.GetMinimalInitrd(ctx, params)
+		Expect(response).Should(BeAssignableToTypeOf(&installer.GetMinimalInitrdNoContent{}))
+	})
+
+	It("returns legit archive", func() {
+		clusterID = strfmt.UUID(uuid.New().String())
+
+		httpProxy := "http://10.10.1.1:3128"
+		httpsProxy := "https://10.10.1.1:3128"
+		noProxy := "quay.io"
+
+		clusterProxyInfo := isoeditor.ClusterProxyInfo{
+			HTTPProxy:  httpProxy,
+			HTTPSProxy: httpsProxy,
+			NoProxy:    noProxy,
+		}
+
+		infraEnv = common.InfraEnv{
+			InfraEnv: models.InfraEnv{
+				ID:            clusterID,
+				PullSecretSet: true,
+				Type:          models.ImageTypeMinimalIso,
+				Proxy: &models.Proxy{
+					HTTPProxy:  &httpProxy,
+					HTTPSProxy: &httpsProxy,
+					NoProxy:    &noProxy,
+				},
+			},
+			PullSecret: "{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}",
```

Comment:
nit: almost worth putting this pull secret in a global var, considering how that it's being copy/pasted throughout this test